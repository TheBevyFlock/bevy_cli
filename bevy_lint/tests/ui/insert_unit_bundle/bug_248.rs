//! This test tracks the bug reported in [#248]. When this starts failing, the bug has been fixed.
//!
//! [#248]: https://github.com/TheBevyFlock/bevy_cli/issues/248

#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::insert_unit_bundle)]

use bevy::prelude::*;

fn main() {
    App::new().add_systems(Startup, my_system);
}

fn my_system(mut commands: Commands) {
    commands.spawn(bundle());
    //~^ ERROR: inserted a `Bundle` containing a unit `()` type
}

fn bundle() -> (Name, ()) {
    // Error should be emitted here, not above.
    (Name::new("Foo"), ())
}
