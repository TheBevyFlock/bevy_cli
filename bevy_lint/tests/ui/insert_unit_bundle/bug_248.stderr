error: inserted a `Bundle` containing a unit `()` type
  --> tests/ui/insert_unit_bundle/bug_248.rs:16:20
   |
16 |     commands.spawn(bundle());
   |                    ^^^^^^^^
   |
   = note: unit `()` types are skipped instead of spawned
note: the lint level is defined here
  --> tests/ui/insert_unit_bundle/bug_248.rs:7:9
   |
7  | #![deny(bevy::insert_unit_bundle)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 1 previous error

